app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -465,13 +465,16 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     )
 
     /**
-     * Caches the last known "is default browser" state when the app was paused.
-     * For an up to do date state use `isDefaultBrowser` instead.
+     * Declared as a function for performance purposes. When this class is first initialized, the
+     * isDefaultBrowser function takes approx. 30-40ms.
      */
-    var wasDefaultBrowserOnLastResume by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_default_browser),
-        default = isDefaultBrowser()
-    )
+    fun checkDefaultBrowserAndSet(): Boolean {
+        val prefKey = appContext.getPreferenceKey(R.string.pref_key_default_browser)
+        val isDefaultBrowserNow = isDefaultBrowser()
+        val isDefaultBrowserBefore = this.preferences.getBoolean(prefKey, isDefaultBrowserNow)
+        this.preferences.edit().putBoolean(prefKey, isDefaultBrowserNow).apply()
+        return isDefaultBrowserNow != isDefaultBrowserBefore
+    }
 
     fun isDefaultBrowser(): Boolean {
```

Comment:
**suggestion (non-blocking):** since we know this is slow, should we change the name to indicate such? e.g. `isDefaultBrowserBlocking` or `isDefaultBrowserSlow`?